### PR TITLE
Add new tests for user service

### PR DESCRIPTION
**Pull request by artemgael96**

### CODE REVIEW

File: exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            _mockHistoryRepository.Setup(x => x.GetHrUserHistoryByIdAsync(It.IsAny<int>())).Returns(UserServiceFixture.GetHrHistoryAsync());
+
+            _mapper = MapperConfigurationProvider.GetConfig().CreateMapper();
+        }
+
+        [Fact]
+        public async Task GetUserHistoryByIdAsync_HistoryModelIsValid()
+        {
+            // Arrange
+            var service = new UserService(_mockUserRepository.Object, _mockHistoryRepository.Object, _mapper);
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs]: add field IUserService _userService = new UserService(_mockUserRepository.Object, _mockHistoryRepository.Object, _mapper);

File: exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            // Assert
+            Assert.NotNull(result);
+            Assert.Equal(50, result[0].Mark);
+        }
+
+        [Fact]
+        public async Task GetHrUserHistoryByIdAsync_HrHistoryModelIsValid()
+        {
+            // Arrange
+            var service = new UserService(_mockUserRepository.Object, _mockHistoryRepository.Object, _mapper);
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs]: take away to field

**Review by ExadelSandbox — COMMENTED** [review]
need work

File: exalearn-api/ExaLearn.Tests/Fixtures/UserServiceFixture.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+﻿using ExaLearn.Dal.Entities;
```
Comment (ExadelSandbox) [inline exalearn-api/ExaLearn.Tests/Fixtures/UserServiceFixture.cs]: delete the file

File: exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        private readonly Mock<IHistoryRepository> _mockHistoryRepository;
+        private readonly IMapper _mapper;
+
+        public UserServiceTests()
+        {
+            _mockUserRepository = new Mock<IUserRepository>();
+            _mockUserRepository.Setup(x => x.GetByIdAsync(It.IsAny<int>())).Returns(UserServiceFixture.GetIdAsync());
+
+            _mockHistoryRepository = new Mock<IHistoryRepository>();
+            _mockHistoryRepository.Setup(x => x.GetUserHistoryByIdAsync(It.IsAny<int>())).Returns(UserServiceFixture.GetListHistoryAsync());
```
Comment (ExadelSandbox) [inline exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs]: remade

**Review by ExadelSandbox — DISMISSED** [review]

File: exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        private readonly Mock<IHistoryRepository> _mockHistoryRepository;
+        private readonly Mock<IMapper> _mockMapper;
+
+        public UserServiceTests()
+        {
+            _mockMapper = new Mock<IMapper>();
+
+            _mockUserRepository = new Mock<IUserRepository>();
+            _mockUserRepository.Setup(x => x.GetByIdAsync(It.IsAny<int>())).Returns(UserServiceFixture.GetIdAsync());
+            //_mockUserRepository.Setup(x => x.GetByIdAsync(It.IsAny<int>()))
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs]: remove comments

File: exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+                        }
+                    });
+                });
+        }
+
+        [Fact]
+        public async Task GetUserHistoryByIdAsync_HistoryModelIsValid()
+        {
+            // Arrange
+            IUserService _userService = new UserService(_mockUserRepository.Object, _mockHistoryRepository.Object, _mockMapper.Object);
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs]: comman field

File: exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+            var history = UserServiceFixture.GetIdAsync();
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs]: spaces between AAA

File: exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+        private readonly IMapper _mapper;
+        private readonly IUserService _userService;
+
+        public UserServiceTests()
+        {
+            _mockUserRepository = new Mock<IUserRepository>();
+            _mockUserRepository.Setup(x => x.GetByIdAsync(It.IsAny<int>()))
+                .Returns(async () =>
+                {
+                    return await new Task<User>(() => new User() { Email = "adminexa@mailnesia.com" });
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs]: Is it necessary?

Comment (romanstat) [reply exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs]: Where start

File: exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+            // Assert
+            Assert.NotNull(result);
+            Assert.Equal(50, result[0].Mark);
+            Assert.Equal(LevelType.Beginner, result[0].Level);
+            Assert.Equal(100, result[1].Mark);
+            Assert.Equal(LevelType.Elementary, result[1].Level);
+        }
+
+        [Fact]
+        public async Task GetHrUserHistoryByIdAsync_HrHistoryModelIsValid()
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs]: Check naming method name_scenario_expected result

Comment (artemgael96) [reply exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs]: fixed

File: exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs (hunk 8 above)
Comment (artemgael96) [reply exalearn-api/ExaLearn.Tests/Services/UserServiceTests.cs]: fixed

**Review by romanstat — APPROVED** [review]

File: exalearn-api/ExaLearn.Bl/Mapping/UserProfile.cs [hunk 10]
```diff
@@ -24,7 +24,7 @@ public UserProfile()
                .ForAllOtherMembers(x => x.Ignore());
 
             CreateMap<PassedTest, HrHistoryDTO>()
-                .ForMember(x => x.FullName, map => map.MapFrom(source => source.User.FirstName + source.User.LastName))
+                .ForMember(x => x.FullName, map => map.MapFrom(source => source.User.FirstName + " " + source.User.LastName))
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/Mapping/UserProfile.cs]: string interpolation

Comment (artemgael96) [reply exalearn-api/ExaLearn.Bl/Mapping/UserProfile.cs]: fixed

**Review by ElenaBurak — APPROVED** [review]

**Pull request merged**